### ISSUE TITLE
component dependancies

### ISSUE THREAD

**Issue by atomicguy** [c-frame/aframe-extras#13]
In trying out different combinations of the extras, I found that Controls requires Math. Which makes complete sense, but the error 'null defined for v.x' threw me for a bit.

Is there a way of requiring/registering one component from another?

Alternatively, a note or dependency graph (I haven't tried other combinations) could be helpful.

**Comment by donmccurdy** [comment]
Great point, thanks! Are you including extras with NPM and `extras.controls.registerAll()`, or cherry-picking files / directories? In either case there's probably a better way I should set this up..

**Comment by atomicguy** [comment]
Tried both options. Though it was an NPM problem for the longest time, so tried just including touch-control.js (the bit I needed for the test at hand) individually. That resulted in a blank scene (I’m pretty sure I referenced it incorrectly).

When I backed up and tried extras.registerAll() everything worked without issue. So I tried just extras.controls.registerAll() and got the error about v.x … and remembered seeing a math component. Added extras.math.registerAll() and was back up and running.
